libs/asset-viewer/src/lib/state/asset-search/asset-search.effects.ts

```diff
@@ -164,53 +88,106 @@ export class AssetSearchEffects {
-          .pipe(map((searchStats) => actions.updateStats({ searchStats })));
-      })
+      ofType(actions.updateSearchResults),
+      withLatestFrom(this.store.select(selectAssetSearchNoActiveFilters)),
+      map(([_, showStudies]) => (showStudies ? actions.closeResults() : actions.openResults()))
     );
   });
 
-  public closeDetailOnUpdateSearch$ = createEffect(() => {
+  public loadStudies$ = createEffect(() => {
     return this.actions$.pipe(
-      ofType(actions.search, actions.resetSearch),
-      map(() => actions.resetAssetDetail())
+      ofType(actions.initializeSearch),
+      switchMap(() => this.store.select(selectStudies)),
+      filter((x) => !x),
+      switchMap(() => this.allStudyService.getAllStudies().pipe(ORD.fromFilteredSuccess)),
+      map((studies) => actions.setStudies({ studies }))
     );
   });
 
-  public setSearchLoadingState$ = createEffect(() => {
-    return this.actions$.pipe(
-      ofType(actions.searchByFilterConfiguration, actions.removePolygon),
-      map(() => actions.setLoadingState())
-    );
-  });
+  /**
+   * Query Parameter Interactions
+   */
+
+  public queryParams$ = this.actions$.pipe(
+    ofType(actions.initializeSearch),
+    first(), // only read query params once
+    concatLatestFrom(() => this.route.queryParams),
+    map(([_, params]) => {
+      const query: AssetSearchQuery = {};
+      const assetId: number | undefined = readNumberParam(params, QUERY_PARAM_MAPPING.assetId);
+      query.text = readStringParam(params, QUERY_PARAM_MAPPING.text);
+      query.polygon = readPolygonParam(params, QUERY_PARAM_MAPPING.polygon);
+      query.authorId = readNumberParam(params, QUERY_PARAM_MAPPING.authorId);
+      const min = readDateParam(params, QUERY_PARAM_MAPPING.createDate.min);
+      const max = readDateParam(params, QUERY_PARAM_MAPPING.createDate.max);
+      query.createDate = min && max ? { min, max } : undefined;
+      query.manCatLabelItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.manCatLabelItemCodes);
+      query.assetKindItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.assetKindItemCodes);
+      query.usageCodes = readArrayParam(params, QUERY_PARAM_MAPPING.usageCodes);
+      query.geometryCodes = readArrayParam(params, QUERY_PARAM_MAPPING.geometryCodes);
+      query.languageItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.languageItemCodes);
+      query.workgroupIds = readArrayParam<number>(params, QUERY_PARAM_MAPPING.workgroupIds);
+      return { query, assetId };
+    }),
+    share()
+  );
+
+  // noinspection JSUnusedGlobalSymbols
+  public readSearchQueryParams$ = createEffect(() =>
+    this.queryParams$.pipe(
+      filter(({ query }) => Object.values(query).some((value) => value !== undefined)),
+      map(({ query }) => actions.search({ query: query }))
+    )
+  );
+
+  public readAssetIdQueryParam$ = createEffect(() =>
+    this.queryParams$.pipe(
+      map(({ assetId }) => assetId),
+      filterNullish(),
```

Comment:
is this cutom method really needed? If so, are there other cases wehre it could be used?